SpatialGDK/ExternalSchemaCodegen/Programs/Improbable.CodeGen.Unreal/HeaderGenerator.cs

```diff
@@ -0,0 +1,376 @@
+            builder.AppendLine("{");
+            builder.AppendLine("public:");
+
+            builder.AppendLine(Text.Indent(1, GenerateUpdateMainDeclarations(type)));
+
+            if (type.Fields.Count + type.Events.Count > 0 )
+            {
+                builder.AppendLine(Text.Indent(1, GenerateOptionalFieldAccessorsDeclarations(name, type, bundle)));
+                builder.AppendLine("private:");
+                builder.AppendLine(Text.Indent(1, GeneratePrivateCommandUpdateFieldDeclarations(type, bundle)));
+            }
+
+            builder.AppendLine("};");
+
+            return builder.ToString().TrimEnd();
+        }
+
+        private static string GenerateUpdateMainDeclarations(TypeDescription type)
+        {
+            var qualifiedType = Types.GetQualifiedTypeName(type);
+            var builder = new StringBuilder();
+
+            builder.AppendLine($@"// Creates a new instance with default values for each field.");
+            builder.AppendLine($@"Update() = default;");
+            builder.AppendLine($@"// Creates a new instance with default values for each field. This is");
+            builder.AppendLine($@"// equivalent to a default-constructed instance.");
+            builder.AppendLine($@"static Update Create() {{ return {{}}; }}");
+            builder.AppendLine($@"// Copyable and movable.");
+            builder.AppendLine($@"Update(Update&&) = default;");
+            builder.AppendLine($@"Update(const Update&) = default;");
+            builder.AppendLine($@"Update& operator=(Update&&) = default;");
+            builder.AppendLine($@"Update& operator=(const Update&) = default;");
+            builder.AppendLine($@"~Update() = default;");
+            builder.AppendLine($@"bool operator==(const Update&) const;");
+            builder.AppendLine($@"bool operator!=(const Update&) const;");
+            builder.Append(Environment.NewLine);
+            builder.AppendLine($@"// Creates an Update from a {qualifiedType} object.");
+            builder.AppendLine($@"static Update FromInitialData(const {qualifiedType}& Data);");
+            builder.Append(Environment.NewLine);
+            builder.AppendLine($@"/**");
+            builder.AppendLine($@"  * Converts to a {qualifiedType}");
+            builder.AppendLine($@"  * object. It is an error to call this function unless *all* of the optional fields in this");
+            builder.AppendLine($@"  * update are filled in.");
+            builder.AppendLine($@"  */");
+            builder.AppendLine($@"{qualifiedType} ToInitialData() const;");
+            builder.Append(Environment.NewLine);
+            builder.AppendLine($@"/**");
+            builder.AppendLine($@"  * Replaces fields in the given {qualifiedType}");
+            builder.AppendLine($@"  * object with the corresponding fields in this update, where present.");
+            builder.AppendLine($@"  */");
+            builder.AppendLine($@"void ApplyTo({qualifiedType}&) const;");
```

Comment:
```suggestion
            builder.AppendLine($@"void ApplyTo(const {qualifiedType}&) const;");
```

Reply:
apply to takes an object of a component type and applies this object to it (updating that objects fields) so should definitely not be const argument